Extension/src/LanguageServer/extension.ts

```diff
@@ -933,6 +1036,79 @@ function onTakeSurvey(): void {
     vscode.commands.executeCommand('vscode.open', uri);
 }
 
+async function onVcpkgOnlineHelpSuggested(): Promise<void> {
+    onActivationEvent();
+    telemetry.logLanguageServerEvent("onVcpkgHelp");
+    let uri: vscode.Uri = vscode.Uri.parse(`https://aka.ms/vcpkg`);
+    vscode.commands.executeCommand('vscode.open', uri);
+}
+
+async function onVcpkgClipboardInstallSuggested(ports?: string[]): Promise<void> {
+    if (!ports) {
+        // Glob up all existing diagnostics for missing includes and look them up in the vcpkg database
+        const missingIncludeLocations: [vscode.TextDocument, vscode.Range[]][] = [];
+        vscode.languages.getDiagnostics().forEach(uriAndDiagnostics => {
+            // Extract textDocument
+            const textDocument: vscode.TextDocument = vscode.workspace.textDocuments.find(doc => doc.uri.fsPath === uriAndDiagnostics[0].fsPath);
+            if (!textDocument) {
+                return;
+            }
+
+            // Extract ranges, taking only those which are for missing includes
+            let ranges: vscode.Range[] = uriAndDiagnostics[1].filter(isMissingIncludeDiagnostic).map<vscode.Range>(d => { return d.range; });
+            if (!ranges.length) {
+                return;
+            }
+
+            // Filter duplicate ranges
+            ranges = ranges.filter((range: vscode.Range, index: number) => {
+                const foundIndex: number = ranges.findIndex(range2 => {
+                    return range.start.line === range2.start.line && range.start.character === range2.start.character &&
+                        range.end.line === range2.end.line && range.end.character === range2.end.character;
+                });
+                return foundIndex === index;
+            });
+
+            missingIncludeLocations.push([textDocument, ranges]);
+        });
+
+        // Queue look ups in the vcpkg database for missing ports; filter out duplicate results
+        let portsPromises: Promise<string[]>[] = [];
+        missingIncludeLocations.forEach(docAndRanges => {
+            docAndRanges[1].forEach(async range => {
+                portsPromises.push(lookupIncludeInVcpkg(docAndRanges[0], range.start.line));
+            });
+        });
+        ports = [].concat(...(await Promise.all(portsPromises)));
+        if (!ports.length) {
+            return Promise.resolve();
+        }
+        ports = ports.filter((port: string, index: number) => { return ports.indexOf(port) === index; });
+    }
+
+    let triplets: string[];
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    switch (platformInfo.platform) {
+        case 'win32':
+            triplets = ['x86-windows', 'x64-windows'];
+            break;
+        case 'darwin':
+            triplets = ['x64-osx'];
+            break;
+        default:
+            triplets = ['x64-linux'];
+    }
+
+    let installCommand: string = 'vcpkg install';
+    ports.forEach(port => {
```

Comment:
Will there always be at least one install to return? Checking if there's a case when it will only return "vcpkg install"

Reply:
`ports` is checked prior, but it is possible that it will be defined but the length is 0. Thanks!